src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs

```diff
@@ -268,12 +268,7 @@ public ViewViewComponentResult View<TModel>(TModel model)
         /// <returns>A <see cref="ViewViewComponentResult"/>.</returns>
         public ViewViewComponentResult View<TModel>(string viewName, TModel model)
         {
-            if (model != null)
-            {
-                ViewData.Model = model;
-            }
-
-            var viewData = new ViewDataDictionary<TModel>(ViewData);
+            var viewData = new ViewDataDictionary<TModel>(ViewData, model);
```

Comment:
@dougbu, could you give me some code example of when this would be an issue? I don't quite understand what you mean.